// react-image-shadow/dist/index.js
'use strict';

var React = require('react');
var classNames = require('classnames');

function _interopDefaultLegacy (e) { return e && typeof e === 'object' && 'default' in e ? e : { 'default': e }; }

var classNames__default = /*#__PURE__*/_interopDefaultLegacy(classNames);

function _defineProperty(obj, key, value) {
  if (key in obj) {
    Object.defineProperty(obj, key, {
      value: value,
      enumerable: true,
      configurable: true,
      writable: true
    });
  } else {
    obj[key] = value;
  }

  return obj;
}

function ownKeys(object, enumerableOnly) {
  var keys = Object.keys(object);

  if (Object.getOwnPropertySymbols) {
    var symbols = Object.getOwnPropertySymbols(object);
    if (enumerableOnly) symbols = symbols.filter(function (sym) {
      return Object.getOwnPropertyDescriptor(object, sym).enumerable;
    });
    keys.push.apply(keys, symbols);
  }

  return keys;
}

function _objectSpread2(target) {
  for (var i = 1; i < arguments.length; i++) {
    var source = arguments[i] != null ? arguments[i] : {};

    if (i % 2) {
      ownKeys(Object(source), true).forEach(function (key) {
        _defineProperty(target, key, source[key]);
      });
    } else if (Object.getOwnPropertyDescriptors) {
      Object.defineProperties(target, Object.getOwnPropertyDescriptors(source));
    } else {
      ownKeys(Object(source)).forEach(function (key) {
        Object.defineProperty(target, key, Object.getOwnPropertyDescriptor(source, key));
      });
    }
  }

  return target;
}

var ImageShadow = function ImageShadow(props) {
  var alt = props.alt,
      className = props.className,
      _props$shadowBlur = props.shadowBlur,
      shadowBlur = _props$shadowBlur === void 0 ? 20 : _props$shadowBlur,
      _props$shadowHover = props.shadowHover,
      shadowHover = _props$shadowHover === void 0 ? false : _props$shadowHover,
      _props$shadowRadius = props.shadowRadius,
      shadowRadius = _props$shadowRadius === void 0 ? 8 : _props$shadowRadius,
      src = props.src,
      style = props.style,
      _props$width = props.width,
      width = _props$width === void 0 ? 300 : _props$width;
  var cls = classNames__default['default']('react-image-shadow', className, {
    'react-image-shadow-hover': shadowHover
  });
  return /*#__PURE__*/React.createElement("div", {
    className: cls,
    style: _objectSpread2({
      width: "".concat(width, "px")
    }, style)
  }, /*#__PURE__*/React.createElement("img", {
    className: "react-image-shadow-img",
    src: src,
    alt: alt,
    width: width,
    style: {
      borderRadius: "".concat(shadowRadius, "px")
    }
  }), /*#__PURE__*/React.createElement("div", {
    className: "react-image-shadow-card",
    style: {
      backgroundImage: "url(".concat(src, ")"),
      borderRadius: "".concat(shadowRadius, "px"),
      filter: "blur(".concat(shadowBlur, "px)")
    }
  }));
};

module.exports = ImageShadow;
